src/clib/pio_darray.c

```diff
@@ -275,10 +275,13 @@ int PIOc_write_darray_multi(int ncid, const int *vid, int ioid, int nvars, PIO_O
  * to.
  * @param ioid the I/O description ID as passed back by
  * PIOc_InitDecomp().
- * @param arraylen the length of the array to be written. This is the
- * length of the local component of the distrubited array. That is,
- * the length of the portion of the data that is on this task. This
- * must be equal to the size of the local array in the decomposition.
+ * @param arraylen the length of the array to be written.
+ * This should be at least the length of the local component of the 
+ * distrubited array. That is, the length of the portion of the data
+ * that is on this task. This must be at least the size of the local
+ * array in the decomposition and the initial elements in the array
```

Comment:
Consider the case were arraylen > number of elements required (as per iodesc) by the current process. In that case we only should be worried about the initial elements (=local ndof) in the array.